internal/rbd/rbd_journal.go

```diff
@@ -405,6 +417,17 @@ func reserveSnap(ctx context.Context, rbdSnap *rbdSnapshot, rbdVol *rbdVolume, c
 		ctx, rbdSnap.JournalPool, journalPoolID, rbdSnap.Pool, imagePoolID,
 		rbdSnap.RequestName, rbdSnap.NamePrefix, rbdVol.RbdImageName, kmsID, rbdSnap.ReservedID, rbdVol.Owner,
 		"", encryptionType)
+	defer func() {
+		// only undo the reservation when an error occurred
+		if err == nil {
+			return
+		}
+
+		undoErr := undoSnapReservation(ctx, rbdSnap, cr)
```

Comment:
This should be and is already called outside after reserveSnap() calls ?

https://github.com/ceph/ceph-csi/blob/devel/internal/rbd/controllerserver.go#L1194
https://github.com/ceph/ceph-csi/blob/devel/internal/rbd/controllerserver.go#L1194

Reply:
That is not sufficient in  case `reserveSnap()` had an internal error. That is the reason why this change was suggested in the 1st place. A failing function should cleanup its own internal partially-completed state on a failure, `reserveSnap()` did not do that yet.